src/node_resource_limits.h

```diff
@@ -0,0 +1,81 @@
+#ifndef SRC_NODE_RESOURCE_LIMITS_H_
+#define SRC_NODE_RESOURCE_LIMITS_H_
+
+#include <uv.h>
+#include <string>
+
+namespace node {
+namespace resource_limits {
+
+/**
+ * Represents OS resource limits for this process.
+ */
+class ResourceLimits {
+ public:
+  virtual ~ResourceLimits() {}
+
+  /**
+   * Get the total amount of memory available to this process.
+   */
+  virtual uint64_t GetMemoryLimitInBytes() {
+    return uv_get_total_memory();
+  }
+};
+
+#ifdef __POSIX__
```

Comment:
Should this actually be restricted to LINUX? Is there another POSIX platform that also offers this?